MoreLinq/ListLike.cs

```diff
@@ -21,34 +21,35 @@ namespace MoreLinq
     using System.Collections.Generic;
 
     /// <summary>
-    /// Represents an list-like (indexable) data structure.
+    /// Represents a union over list types implementing either
+    /// <see cref="IList{T}"/> or <see cref="IReadOnlyList{T}"/>, allowing
+    /// both to be treated the same.
     /// </summary>
 
-    interface IListLike<out T>
+    struct ListLike<T> // TODO Can be readonly when using C# 7.2
```

Comment:
I believe this TODO is now actionable as 7.3 is the minimum version https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version

Reply:
Addressed with 659a56b40061de6496260b6f17062d1ec3e2a160.